File: day3/solution.js
import fs from "fs";

const banks = fs.readFileSync("input.txt", "utf-8").trim().split("\n");
let total = 0;
for (const bank of banks) {
  const length = bank.length;
  let max = 0;
  let maxLeft = 0;
  for (let i = 0; i < length - 1; i++) {
    if (parseInt(bank[i]) >= maxLeft) {
      let maxRight = 0;
      maxLeft = parseInt(bank[i]);
      for (let j = i + 1; j < length; j++) {
        if (parseInt(bank[j]) > maxRight) {
          maxRight = parseInt(bank[j]);
        }
      }
      if (maxLeft * 10 + maxRight > max) {
        max = maxLeft * 10 + maxRight;
      }
    }
  }
  total += max;
}
console.log(total);

File: day3/solution.py
with open("input.txt", "r") as f:
    banks = f.read().strip().split("\n")

total = 0
for bank in banks:
    length = len(bank)
    max_val = 0
    max_left = 0
    for i in range(length - 1):
        if int(bank[i]) > max_left:
            max_right = 0
            max_left = int(bank[i])
            for j in range(i + 1, length):
                if int(bank[j]) > max_right:
                    max_right = int(bank[j])
            if max_left * 10 + max_right > max_val:
                max_val = max_left * 10 + max_right
    total += max_val

print(total)
